pkg/appgw/test_fixtures.go

```diff
@@ -25,13 +35,49 @@ func makeConfigBuilderTestFixture() appGwConfigBuilder {
 			},
 			CertificateSecretStore: nil,
 		},
-		secretIDCertificateMap: make(map[secretIdentifier]*string),
 	}
+
+	return cb
+}
+
+func addCertsTestFixture(cb *appGwConfigBuilder) {
+	ingressKey := "--ingress-key--"
+	hostName := "--some-hostname--"
+	secretsIdent := secretIdentifier{
+		Namespace: "--namespace--",
+		Name:      "--name--",
+	}
+	cb.ingressKeyHostnameSecretIDMap[ingressKey] = make(map[string]secretIdentifier)
+	cb.ingressKeyHostnameSecretIDMap[ingressKey][hostName] = secretsIdent
+	// Wild card
+	cb.ingressKeyHostnameSecretIDMap[ingressKey][""] = secretsIdent
+
+	cb.secretIDCertificateMap[secretsIdent] = to.StringPtr("")
 }
 
 func makeIngressTestFixture() v1beta1.Ingress {
 	return v1beta1.Ingress{
 		Spec: v1beta1.IngressSpec{
+			Rules: []v1beta1.IngressRule{
+				{
+					Host: "-some-host-",
+					IngressRuleValue: v1beta1.IngressRuleValue{
+						HTTP: &v1beta1.HTTPIngressRuleValue{
+							Paths: []v1beta1.HTTPIngressPath{
+								{
+									Path: "////",
```

Comment:
/path

Reply:
I'm going to add a path shortly. (Haven't figured out what makes sense here just yet, bit it is coming!)

Thanks for the quick review!!